packages/network-controller/src/NetworkController.ts

```diff
@@ -730,17 +730,23 @@ export class NetworkController extends BaseControllerV2<
   /**
    * Fetches the network ID for the network, ensuring that it is a hex string.
    *
+   * @param networkClientId - The ID of the network client to fetch the network
    * @returns A promise that either resolves to the network ID, or rejects with
    * an error.
    * @throws If the network ID of the network is not a valid hex string.
    */
-  async #getNetworkId(): Promise<NetworkId> {
+  async #getNetworkId(networkClientId?: NetworkClientId): Promise<NetworkId> {
```

Comment:
Similar point as below, but what do you think about having this method always take a network client ID?